homeassistant/components/tuya/config_flow.py

```diff
@@ -16,131 +17,108 @@
     CONF_ENDPOINT,
     CONF_PASSWORD,
     CONF_PROJECT_TYPE,
+    CONF_REGION,
     CONF_USERNAME,
     DOMAIN,
-    TUYA_APP_TYPE,
-    TUYA_ENDPOINT,
-    TUYA_PROJECT_TYPE,
+    SMARTLIFE_APP,
+    TUYA_REGIONS,
+    TUYA_RESPONSE_CODE,
+    TUYA_RESPONSE_MSG,
+    TUYA_RESPONSE_PLATFROM_URL,
+    TUYA_RESPONSE_RESULT,
+    TUYA_RESPONSE_SUCCESS,
+    TUYA_SMART_APP,
 )
 
-RESULT_SINGLE_INSTANCE = "single_instance_allowed"
-RESULT_AUTH_FAILED = "invalid_auth"
-TUYA_ENDPOINT_BASE = "https://openapi.tuyacn.com"
-TUYA_ENDPOINT_OTHER = "https://openapi.tuyaus.com"
-COUNTRY_CODE_CHINA = ["86", "+86", "China"]
-
 _LOGGER = logging.getLogger(__name__)
 
-# Project Type
-DATA_SCHEMA_PROJECT_TYPE = vol.Schema(
-    {vol.Required(CONF_PROJECT_TYPE, default=0): vol.In(TUYA_PROJECT_TYPE)}
-)
-
-# INDUSTY_SOLUTIONS Schema
-DATA_SCHEMA_INDUSTRY_SOLUTIONS = vol.Schema(
-    {
-        vol.Required(CONF_ENDPOINT): vol.In(TUYA_ENDPOINT),
-        vol.Required(CONF_ACCESS_ID): str,
-        vol.Required(CONF_ACCESS_SECRET): str,
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-    }
-)
-
-# SMART_HOME Schema
-DATA_SCHEMA_SMART_HOME = vol.Schema(
-    {
-        vol.Required(CONF_ACCESS_ID): str,
-        vol.Required(CONF_ACCESS_SECRET): str,
-        vol.Required(CONF_APP_TYPE): vol.In(TUYA_APP_TYPE),
-        vol.Required(CONF_COUNTRY_CODE): str,
-        vol.Required(CONF_USERNAME): str,
-        vol.Required(CONF_PASSWORD): str,
-    }
-)
-
 
 class TuyaConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Tuya Config Flow."""
 
-    def __init__(self) -> None:
-        """Init tuya config flow."""
-        super().__init__()
-        self.conf_project_type = None
-
     @staticmethod
-    def _try_login(user_input):
-        project_type = ProjectType(user_input[CONF_PROJECT_TYPE])
-        api = TuyaOpenAPI(
-            user_input[CONF_ENDPOINT]
-            if project_type == ProjectType.INDUSTY_SOLUTIONS
-            else "",
-            user_input[CONF_ACCESS_ID],
-            user_input[CONF_ACCESS_SECRET],
-            project_type,
-        )
-        api.set_dev_channel("hass")
-
-        if project_type == ProjectType.INDUSTY_SOLUTIONS:
-            response = api.login(user_input[CONF_USERNAME], user_input[CONF_PASSWORD])
-        else:
-            if user_input[CONF_COUNTRY_CODE] in COUNTRY_CODE_CHINA:
-                api.endpoint = TUYA_ENDPOINT_BASE
+    def _try_login(user_input: dict[str, Any]) -> tuple[dict[Any, Any], dict[str, Any]]:
+        """Try login."""
+        response = {}
+
+        data = {
+            CONF_ENDPOINT: TUYA_REGIONS[user_input[CONF_REGION]],
+            CONF_PROJECT_TYPE: ProjectType.INDUSTY_SOLUTIONS,
+            CONF_ACCESS_ID: user_input.get(CONF_ACCESS_ID),
+            CONF_ACCESS_SECRET: user_input.get(CONF_ACCESS_SECRET),
+            CONF_USERNAME: user_input.get(CONF_USERNAME),
+            CONF_PASSWORD: user_input.get(CONF_PASSWORD),
```

Comment:
All the input is required.
```suggestion
            CONF_ACCESS_ID: user_input[CONF_ACCESS_ID],
            CONF_ACCESS_SECRET: user_input[CONF_ACCESS_SECRET],
            CONF_USERNAME: user_input[CONF_USERNAME],
            CONF_PASSWORD: user_input[CONF_PASSWORD],
```